src/plugins/saved_objects_finder/public/finder/saved_object_finder.test.tsx

```diff
@@ -217,6 +219,29 @@ describe('SavedObjectsFinder', () => {
         </React.Fragment>
       `);
     });
+
+    it('render description if provided', async () => {
```

Comment:
In this test you are testing implementation details which is not ideal. Changing the implementation (underlying component, props to those component) will break this test.

Concretely you assume that the `EuiInMemoryTable` is present and that it takes `items` prop.

Now I know that you basically copied what other tests are doing 😊  but let's take this opportunity to improve the test.

In https://github.com/elastic/kibana/blob/main/packages/kbn-test-jest-helpers/src/testbed/testbed.ts I wrote a helper function to parse an EUI table and read its content for testing purpose. Let's copy it over in this file.

So on L37 (below the imports), add the helper func

```ts
...
import { coreMock } from '@kbn/core/public/mocks';

// Copied from packages/kbn-test-jest-helpers/src/testbed/testbed.ts
const getTableMetadata = (subject: string, reactWrapper: ReactWrapper) => {
  const table = findTestSubject(reactWrapper, subject);

  if (!table.length) {
    throw new Error(`Eui Table "${subject}" not found.`);
  }

  const rows = table
    .find('tr')
    .slice(1) // we remove the first row as it is the table header
    .map((row) => ({
      reactWrapper: row,
      columns: row.find('div.euiTableCellContent').map((col) => ({
        reactWrapper: col,
        // We can't access the td value with col.text() because
        // eui adds an extra div in td on mobile => (.euiTableRowCell__mobileHeader)
        value: col.find('div.euiTableCellContent').text(),
      })),
    }));

  // Also output the raw cell values, in the following format: [[td0, td1, td2], [td0, td1, td2]]
  const tableCellsValues = rows.map(({ columns }) => columns.map((col) => col.value));
  return { rows, tableCellsValues };
};
```

Now this test can become

```ts
it('render description if provided', async () => {
  (contentClient.mSearch as any as jest.SpyInstance).mockImplementation(() =>
    Promise.resolve({ hits: [doc, doc2, doc4] })
  );

  const wrapper = mount(
    <SavedObjectFinder
      services={{ uiSettings, contentClient, savedObjectsTagging }}
      savedObjectMetaData={searchMetaData}
    />
  );

  await nextTick();
  wrapper.update();

  const table = getTableMetadata('savedObjectsFinderTable', wrapper);
  expect(table.tableCellsValues).toEqual([
    ['Another titleanother description', 'tag-2'],
    ['Example titleexample description', 'tag-1'],
    ['Search', 'tag-4'],
  ]);
});
```

See how we don't test the underlying implementation. We test that the table renders the correct text.